drivers/spi/spi_nrfx_spi.c

```diff
@@ -277,8 +291,13 @@ static int spi_nrfx_pm_action(const struct device *dev,
 
 	switch (action) {
 	case PM_DEVICE_ACTION_RESUME:
-		/* No action needed at this point, nrfx_spi_init() will be
-		 * called at configuration before the next transfer.
+		ret = spi_nrfx_pins_init(dev);
+		if (ret < 0) {
+			return ret;
+		}
+
+		/* nrfx_spi_init() will be called at configuration before
+		 * the next transfer.
```

Comment:
why do who have a helper here? I'd be consistent and use #ifdefs everywhere, as in the sleep case.

Reply:
Default configuration of the pins needs to be also applied when the driver instance is initialized (in `spi_##idx##_init()`). That's why `spi_nrfx_pins_init()` was introduced.

Reply:
why can't we use regular ifdeffed pinctrl calls? `spi_nrfx_pins_init` naming is confusing IMO, and inconsistent with the sleep counterpart.

Reply:
I was afraid that `spi_##idx##_init()`, which is in a macro, would be difficult to read then. But I agree, it is confusing it the current form. I'll get rid of `spi_nrfx_pins_init()`.